var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -470,6 +470,15 @@ def define_enable(suffix, value=None):
                                    'explicit_template_instantiation')
         ])
 
+        # Suppress TriBITS flags in favor of CMake's built-in flags
+        if spec.version < Version('13'):
+            # Prior to version 13, Trilinos would erroneously inject
+            # '-std=c++11' regardless of CMAKE_CXX_STANDARD value
+            options.append(
+                define('Trilinos_CXX11_FLAGS',
+                       '-std=c++' + spec.variants['cxxstd'].value)
+            )
```

Comment:
I think this is good (though there's certainly some floor to this working as well, and who knows if it's one in the spack version). I suggest removing the always-on line above this:
```
            define_from_variant('CMAKE_CXX_STANDARD', 'cxxstd'),
```
and combine it with your conditional:
```python
        if spec.version >= Version('13'):
             options.append(define_from_variant('CMAKE_CXX_STANDARD', 'cxxstd'))
        else:
             # Prior to version 13, Trilinos would erroneously inject
             # '-std=c++11' regardless of CMAKE_CXX_STANDARD value
             options.append(
                 define('Trilinos_CXX11_FLAGS',
                        '-std=c++' + spec.variants['cxxstd'].value)
             )
```

Reply:
I like that solution. I'm testing it out and I'll push it as soon as I try it out.

Reply:
The [above conditional](https://github.com/spack/spack/pull/28663/files#r795030899) looks correct, assuming the variant `cxxstd` takes on values `11`, `14`, `17`, `20`, etc.